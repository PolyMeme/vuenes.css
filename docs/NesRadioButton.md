# nes-radio-button 

## slots 

- `default` 

## props 

- `group-name` ***String*** (*required*) 

