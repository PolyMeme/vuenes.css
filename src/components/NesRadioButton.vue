<template>
  <label>
    <input
      class="nes-radio"
      type="radio"
      :name="groupName"
      :checked="checked"
    >
    <span>
      <!-- Default contents of the Radio button -->
      <slot />
    </span>
  </label>
</template>

<script>
export default {
  props: {
    /**
     * Define the `name` of the radio button
     */
    groupName: {
      type: String,
      required: true
    },
    /**
     * Define if the radio button is checked
     */
    checked: {
      type: Boolean,
      required: false
    }
  }
}
</script>
