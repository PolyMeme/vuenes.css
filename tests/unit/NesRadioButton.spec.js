import { mount } from '@vue/test-utils'
import NesRadioButton from '../../src/components/NesRadioButton.vue'

describe('NesRadioButton.vue', () => {
  const createRadioButtonWrapper = (propsData = {}, slot) => {
    return mount(NesRadioButton, {
      propsData,
      slots: {
        default: slot || 'I am a radio button!'
      }
    })
  }

  it('should render component properly', () => {
    const wrapper = createRadioButtonWrapper({
      groupName: 'testgroup'
    })

    const radioInput = wrapper.find('label>input[type="radio"][name="testgroup"].nes-radio')
    expect(radioInput.exists()).toBe(true)
    expect(wrapper.find('label>span').text()).toEqual('I am a radio button!')
  })

  it('should render slot properly', () => {
    const wrapper = createRadioButtonWrapper({
      groupName: 'testgroup'
    }, 'This is a text in a slot')

    const slot = wrapper.find('label>span')
    expect(slot.exists()).toBe(true)
    expect(slot.text()).toEqual('This is a text in a slot')
  })

  it('should match snapshot', () => {
    const wrapper = createRadioButtonWrapper({
      groupName: 'testgroup'
    }, 'This is a text in a slot')

    expect(wrapper.html()).toMatchSnapshot()
  })
})
