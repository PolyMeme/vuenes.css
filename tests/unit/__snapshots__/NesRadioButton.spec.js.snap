// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NesRadioButton.vue should match snapshot 1`] = `<label><input type="radio" name="testgroup" class="radio"> <span>This is a text in a slot</span></label>`;
